core/base/src/TString.cxx

```diff
@@ -361,7 +375,8 @@ TString& TString::operator=(const TSubString &substr)
 
 TString& TString::Append(char c, Ssiz_t rep)
 {
-   if (!rep) return *this;
+   if (rep <= 0 ) return *this; // appends nothing
```

Comment:
Split to two lines, otherwise when ran in a debugger, one cannot distinguish the if clause from the statement.